azure_translator/__init__.py

```diff
@@ -0,0 +1,89 @@
+"""Azure Translator module."""
+import xml.etree.ElementTree as ET
+
+import requests
+
+
+class Translator(object):
+    """
+    Translator class, connects to Azure Translator API:
+
+        http://docs.microsofttranslator.com/text-translate.html
+    """
+    DEFAULT_LANGUAGE = 'en'
+    TOKEN_API = 'https://api.cognitive.microsoft.com/sts/v1.0/issueToken'
+    TRANSLATE_API = 'https://api.microsofttranslator.com/v2/http.svc/Translate'
+
+
+    def __init__(self, api_key):
+        self.api_key = api_key
+
+    @staticmethod
+    def _api_error(request_error):
+        return AzureApiError(
+            unicode(request_error),
+            response=request_error.response,
+            request=request_error.request
+        )
+
+    def get_access_token(self):
+        """
+        Retrieve an access token in order to use the Translator API.
+        """
+        resp = requests.post(
+            self.TOKEN_API,
+            headers={
+                'Content-Type': 'application/json',
+                'Accept': 'application/jwt',
+                'Ocp-Apim-Subscription-Key': self.api_key,
+            }
+        )
+        try:
+            resp.raise_for_status()
+        except requests.exceptions.HTTPError as error:
+            raise self._api_error(error)
+        return resp.content
+
+    def translate(self, text, to=DEFAULT_LANGUAGE):
+        """
+        Translate some text into some language. Target language default to english.
+
+        :param text:               the text to translate
+        :param to:                 the target language
+        """
+        resp = requests.get(
+            self.TRANSLATE_API,
+            headers={
+                'Authorization': 'Bearer {}'.format(self.get_access_token()),
```

Comment:
I could enhance it and get the token if it expired, but I didn't want to rely on an exception from Azure API, because I don't know how consistent is this API yet...